### PR TITLE
Face editor add eye gaze & remove open factor sliders

### PR DESCRIPTION
**Pull request by harisreedhar**

### CODE REVIEW

File: facefusion.ini [hunk 1]
```diff
@@ -52,9 +52,7 @@ expression_restorer_factor =
 face_debugger_items =
 face_editor_model =
 face_editor_eye_open_ratio =
```
Comment (henryruhs) [inline facefusion.ini:54]: you forgot to add both gaze options

File: facefusion/processors/modules/face_editor.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 from facefusion.execution import create_inference_pool
 from facefusion.face_analyser import get_many_faces, get_one_face
 from facefusion.face_helper import paste_back, warp_face_by_face_landmark_5
-from facefusion.face_masker import create_face_mask, create_occlusion_mask, create_static_box_mask
+from facefusion.face_masker import create_occlusion_mask, create_static_box_mask
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:18]: can we delete create_face_mask then?

File: facefusion/processors/modules/face_editor.py [hunk 3]
```diff
@@ -169,7 +169,9 @@ def post_process() -> None:
 def edit_face(target_face: Face, temp_vision_frame : VisionFrame) -> VisionFrame:
 	model_template = get_model_options().get('template')
 	model_size = get_model_options().get('size')
-	crop_vision_frame, affine_matrix = warp_face_by_face_landmark_5(temp_vision_frame, target_face.landmark_set.get('5/68'), model_template, model_size)
+	face_landmark_5 = target_face.landmark_set.get('5/68')
+	face_landmark_5 = (face_landmark_5 - face_landmark_5[2]) * 1.2 + face_landmark_5[2]
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:173]: can you explain this 1.2 factor or in general what you are doing here?

File: facefusion/processors/modules/face_editor.py [hunk 4]
```diff
@@ -205,26 +201,16 @@ def apply_edit_face(crop_vision_frame : VisionFrame, face_landmark_68 : FaceLand
 		})[0]
 
 	with thread_semaphore():
-		motion_points = motion_extractor.run(None,
+		rotation, scale, translation, expression, motion_points_edit, motion_points = motion_extractor.run(None,
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:204]: the inputs is named angle, scale, translation, expression, motions_points_raw, motion_points - please follow that naming

File: facefusion/processors/modules/face_editor.py [hunk 5]
```diff
@@ -233,29 +219,81 @@ def apply_edit_face(crop_vision_frame : VisionFrame, face_landmark_68 : FaceLand
...
+	left_eye_ratio = calc_distance_ratio(face_landmark_68, 37, 40, 39, 36)
+	right_eye_ratio = calc_distance_ratio(face_landmark_68, 43, 46, 45, 42)
+
+	if face_editor_eye_open_ratio < 0:
+		close_eye_ratio = numpy.array([ left_eye_ratio, right_eye_ratio, 0 ]).astype(numpy.float32).reshape(1, -1)
+
+		with thread_semaphore():
+			close_eye_motion_points = eye_retargeter.run(None,
+			{
+				'input': numpy.concatenate([ motion_points.reshape(1, -1), close_eye_ratio ], axis = 1)
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:256]: do the concatenate outside of the semaphore block: potential better performance (I had similar issues before) but for sure better readability

File: facefusion/processors/modules/face_editor.py [hunk 6]
```diff
@@ -233,29 +219,81 @@ def apply_edit_face(crop_vision_frame : VisionFrame, face_landmark_68 : FaceLand
...
-	return eye_ratio, lip_ratio
+	lip_ratio = calc_distance_ratio(face_landmark_68, 62, 66, 54, 48)
+
+	if face_editor_lip_open_ratio < 0:
+		close_lip_ratio = numpy.array([ lip_ratio, 0 ]).astype(numpy.float32).reshape(1, -1)
+
+		with thread_semaphore():
+			close_lip_motion_points = lip_retargeter.run(None,
+			{
+				'input': numpy.concatenate([ motion_points.reshape(1, -1), close_lip_ratio ], axis = 1)
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:283]: do the reshape of motion_points not within semaphore block

File: facefusion/wording.py [hunk 7]
```diff
@@ -136,6 +136,11 @@
 		'expression_restorer_model': 'choose the model responsible for restoring the expression',
 		'expression_restorer_factor': 'restore factor of expression from target face',
 		'face_debugger_items': 'load a single or multiple processors (choices: {choices})',
+		'face_editor_model': 'choose the model responsible for editing the face',
```
Comment (henryruhs) [inline facefusion/wording.py:139]: editing the faces (with s)

File: facefusion/processors/modules/face_editor.py [hunk 8]
```diff
@@ -233,29 +219,81 @@ def apply_edit_face(crop_vision_frame : VisionFrame, face_landmark_68 : FaceLand
...
+	expression[0, 1, 1] += map_float(face_editor_eye_gaze_vertical, -1, 1, -0.0025, 0.0025)
+	expression[0, 2, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.0025, 0.0025)
+	expression[0, 11, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.010, 0.010)
+	expression[0, 13, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.005, 0.005)
+	expression[0, 15, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.010, 0.010)
+	expression[0, 16, 1] -= map_float(face_editor_eye_gaze_vertical, -1, 1, -0.005, 0.005)
+	return expression
+
+
+def edit_eye_open(motion_points : NDArray[Any], face_landmark_68 : FaceLandmark68) -> NDArray[Any]:
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:244]: Replace the NDArray[Any] types with speaking names

File: facefusion/processors/modules/face_editor.py [hunk 9]
```diff
@@ -233,29 +219,81 @@ def apply_edit_face(crop_vision_frame : VisionFrame, face_landmark_68 : FaceLand
...
+	eye_retargeter = get_inference_pool().get('eye_retargeter')
 	face_editor_eye_open_ratio = state_manager.get_item('face_editor_eye_open_ratio')
+	left_eye_ratio = calc_distance_ratio(face_landmark_68, 37, 40, 39, 36)
+	right_eye_ratio = calc_distance_ratio(face_landmark_68, 43, 46, 45, 42)
+
+	if face_editor_eye_open_ratio < 0:
+		close_eye_ratio = numpy.array([ left_eye_ratio, right_eye_ratio, 0 ]).astype(numpy.float32).reshape(1, -1)
```
Comment (henryruhs) [inline facefusion/processors/modules/face_editor.py:251]: super minor detail but try to add astype casting to the end always

**Review by henryruhs — CHANGES_REQUESTED** [review]

File: facefusion/processors/modules/face_editor.py (hunk 4 above)
Comment (harisreedhar) [reply facefusion/processors/modules/face_editor.py:206]: it is not really angle, but a rotation matrix

File: facefusion/processors/modules/face_editor.py (hunk 2 above)
Comment (harisreedhar) [reply facefusion/processors/modules/face_editor.py:18]: It is only needed for expression-restorer. so far face-editor doesn't needs it.

File: facefusion/processors/modules/face_editor.py (hunk 4 above)
Comment (henryruhs) [reply facefusion/processors/modules/face_editor.py:206]: then we have to rename it to rotation_matrix

File: facefusion/processors/modules/face_editor.py (hunk 3 above)
Comment (harisreedhar) [reply facefusion/processors/modules/face_editor.py:175]: scaling face-landmark from the center, face_landmark_5[2] is the nose which is considered as center.

**Review by henryruhs — APPROVED** [review]

**Pull request merged**